### PR TITLE
Adding GitHub calendar based on username

### PR DESCRIPTION
**Pull request by gvicas17**
## PR TITLE

**Description:** 

  
Action for adding github username was added to actions file
GitHub username settings card was added to SettingsContent.js using the action for adding a username
react-github-calendar was added to ProfileContent.js and props from github username passed in for specific user to display their calendar
  
  
Github calendar now shows up if user has added Github username to their profile settings
  

### Checklist

 * [x] My code follows the style guidelines of this project
 * [x] I have performed a self-review of my own code
 * [ ] My code has been reviewed by at least one peer
 * [ ] I have commented my code, particularly in hard-to-understand areas
 * [x] I have made corresponding changes to the documentation
 * [x] My changes generate no new warnings
 * [ ] I have added tests that prove my fix is effective or that my feature works
 * [ ] New and existing unit tests pass locally with my changes
 * [x] There are no merge conflicts

### CODE REVIEW

File: src/assets/ProfileContent.js [hunk 1]
```diff
@@ -4,9 +4,13 @@ import { connect } from 'react-redux';
...
 import SettingsContent from './SettingsContent';
 import { Layout, Badge, Button, Tabs, Card } from 'antd';
-import Column from 'antd/lib/table/Column';
+import GitHubCalendar from 'react-github-calendar'
+
+
+
 
 const ProfileContent = (props) => {
+  console.log(props)
```
Comment (Plofland) [inline src/assets/ProfileContent.js:13]: console.log here

File: src/assets/SettingsContent.js [hunk 2]
```diff
@@ -20,21 +21,28 @@ const SettingsContent = (props) => {
...
   const [editSettings, setEditSettings] = useState(initialEditSettings);
 
   const [input, setInput] = useState('');
+  const [github, setGithub] = useState('')
 
   useEffect(() => {
     setSettings({ ...settings, displayName: props.user.displayName });
   }, [props.user]);
 
+  useEffect(() => {
```
Comment (Plofland) [inline src/assets/SettingsContent.js:42]: We can combine this with the above useEffect.

Comment (Plofland) [inline src/assets/SettingsContent.js]: We'll combine these [props.user] useEffect.

File: src/store/actions/index.js [hunk 3]
```diff
@@ -526,6 +526,24 @@ export const fetchComments = (commentId) => (dispatch) => {
...
+
+export const updateGitHubUsername = (userDetails, gitHubUsername) => (
+  dispatch
+) => {
+    console.log(userDetails, gitHubUsername);
+  const userID = userDetails.id;
+  axiosWithAuth()
+    .put(`${BACKEND_URL}/api/user/githubusername`, { userID, gitHubUsername })
+    .then(() => {
+      toast.success('Woo! Username changed to ' + gitHubUsername);
```
Comment (Plofland) [inline src/store/actions/index.js]: Minor change. 'Woo! Username changed to'  --> 'Woo! GitHub username changed to'

File: src/store/actions/index.js [hunk 4]
```diff
@@ -526,6 +526,24 @@ export const fetchComments = (commentId) => (dispatch) => {
...
+      dispatch({ type: 'SET_USER', payload: { ...userDetails, gitHubUsername } });
+    })
+    .catch(() =>
+      toast.error('Oh no! there was a problem updating your username.')
```
Comment (Plofland) [inline src/store/actions/index.js]: Same change here.

**Review by Plofland — CHANGES_REQUESTED** [review]
Very minor changes.

**Review by ialkamal — APPROVED** [review]
APPROVED

File: src/assets/ProfileContent.js (hunk 1 above)
Comment (Plofland) [reply src/assets/ProfileContent.js:13]: Thanks

File: src/store/actions/index.js (hunk 3 above)
Comment (Plofland) [reply src/store/actions/index.js]: Thanks

File: src/store/actions/index.js (hunk 4 above)
Comment (Plofland) [reply src/store/actions/index.js]: Thanks

File: src/assets/SettingsContent.js (hunk 2 above)
Comment (Plofland) [reply src/assets/SettingsContent.js]: Thanks

**Review by robo-chick — APPROVED** [review]
Approved!

**Review by Plofland — APPROVED** [review]

**Pull request merged**